### ISSUE TITLE
Mail::Field::ParseError

### ISSUE THREAD

**Issue by zankevich** [mikel/mail#995]
```
Mail::Field::ParseError: Mail::AddressList can not parse |"Yelp – Fort Worth" <no-reply@yelp.com>|
Reason was: Only able to parse up to "Yelp –
```

**Comment by justin-yesware** [comment]
Are you using `Mail::Address.new(string)`? If so, this requires that the string be encoded so any non-supported characters (in your case, the dash) is marked up. I've found two ways to get around this:
```
Mail::ToField.new(s) => Mail::ToField
# or
Mail::Address.new(Mail::Encodings.decode_encode(s, :encode)) => String
```

I'd be interested in hearing if there's a better way to deal with addresses that come in like this (not encoded).

**Comment by jwg2s** [comment]
Also getting this error, but using @justin-yesware approach `Mail::AddressList.new` results in incorrectly parsed headers.

Any ideas on a more graceful fix?  Seems like something that needs to happen down at a lower level like in `Mail::Parsers`.  This is a valid UTF-8 character that we're dying on.

What does `Mail::ToField` do that's different than what `Mail::AddressList` does?

**Comment by jwg2s** [comment]
@zankevich how was this resolved?

**Comment by jwg2s** [comment]
😆 I'll try to find some time to look at what those two are doing differently and open a PR that'll address the issue, no pun intended

**Comment by jeremy** [comment]
Fixed by #1103
